pkg/querier/blocks_store_replicated_set.go

```diff
@@ -118,12 +123,19 @@ func (s *blocksStoreReplicationSet) GetClientsFor(userID string, blockIDs []ulid
 		}
 
 		// Pick a non excluded store-gateway instance.
-		addr := getNonExcludedInstanceAddr(set, exclude[blockID], s.balancingStrategy)
-		if addr == "" {
+		instance := getNonExcludedInstanceAddr(set, exclude[blockID], s.balancingStrategy, s.zoneAwarenessEnabled, attemptedBlocksZones[blockID])
```

Comment:
nit: rename to `getNonExcludedInstance`. It no longer returns `Addr`

Reply:
Fixed